### ISSUE TITLE
Table: Allow table cells to calculate min and max based on column's values

### ISSUE THREAD

**Issue by jutley** [grafana/grafana#26383]
**What would you like to be added**:

When using table cells with gauge displays, I would like the automatically calculated "max" to be based on the values in the cells column, not the table as a whole.

**Why is this needed**:

In any table, it is reasonable to expect the values in each column to represent completely different concepts. For example, we may want each row to represent a Kubernetes Pod, and have columns to show CPU usage and memory usage. These two values are drastically different in magnitude. CPU is measured in cores, and may be in the range of 0-4. Memory is measured in bytes, and may the range of 0-4*1024^3. 

This means that the automatically calculated "Max" may be 4*1024^3. A gauge representing CPU usage will look identical for both very low CPU (0.01) and very high CPU (4). This makes the bar gauge functionally useless in this context. 

Right now, the best we can do is to define the max manually, but this is very finicky. If set too low, the gauges will look identical for any value above the max, which is misleading. If it is too high, it tends to waste a lot of horizontal space. Plus, the max in one context may be completely inappropriate for another context.

**Example**:

Here is an example that illustrated what is described above. You can see that while the CPU Usage values differ by orders of magnitude, they all appear exactly the same. This is because the very large values for "Memory Usage" are determining the max value for the "CPU Usage" bar gauges.

![image](https://user-images.githubusercontent.com/5660346/91223363-5ebba600-e6d5-11ea-82b0-86d3f6eb958f.png)

**Comment by weibeld** [comment]
This feature would be extremely valuable. Currently, we're effectively restricted to a single column per table with numerical values when using the gauge display mode (unless the values in all columns are in a similar range, which is often not the case). So, I hope this feature will actually make it to the 7.4.0 release.

**Comment by torkelo** [comment]
@ryantxu currently the BarGaugeCell calls getFieldConfigWithMinMax with local: false parameter, we can fix this issue and get local (field) level min & max for bar gauge in table. 

Feels like the we can just change that , but the question is should there be an option for this and is this an ok breaking change? My thinking is yes.

**Comment by MAFLO321** [comment]
Any updates on this?

**Comment by MAFLO321** [comment]
@torkelo @ivanahuckova Any updates on this?

**Comment by MAFLO321** [comment]
@torkelo @ivanahuckova Any updates on this? Maybe move to a different milestone.

**Comment by imReker** [comment]
😢  Still a problem on Grafana v9.0, maybe add to v100.0 milestone?
![image](https://user-images.githubusercontent.com/4949937/180124786-8de9e37d-a967-4e0e-91a2-757cc5910bdf.png)

**Comment by mdvictor** [comment]
We are currently looking into this. Table cell types have been recently revisited and the refactor will make it easier to fix this.

**Comment by Legion2** [comment]
![image](https://user-images.githubusercontent.com/12752323/234991795-ee51da3a-d4a7-489c-af3c-050830b90c9e.png)
@sonium0 Do you mean these Min/Max, I think they exist for a long time now and don't solve the problem.

**Comment by sonium0** [comment]
No, as I said if you specify the min and max as the OVERWRITE, I found this solves the problem On 27. Apr 2023, at 23:11, Leon Kiefer ***@***.***> wrote:﻿

@sonium0 Do you mean these Min/Max, I think they exist for a long time now and don't solve the problem.

**Comment by sonium0** [comment]
No, as I said if you specify the min and max as the OVERWRITE, I found this solves the problem

**Comment by Legion2** [comment]
But I don't know what the min max is, so I want to use the "auto" setting, which doesn't work for multiple columns.

**Comment by tomasgatial** [comment]
Still broken as of v9.5.0-cloud.5.a016665c 
Each column uses min/max value from table context. 

My use case:
<img width="1710" alt="Screenshot 2023-04-28 at 10 00 50" src="https://user-images.githubusercontent.com/3351584/235090617-e3382bf5-080c-471f-924b-e4acf1788c1a.png">

**Comment by CostinSM** [comment]
I am currently on Grafana v10.0.0-cloud.3.b04cc88b and still no sign of this request being implemented
Are there any news about it?

**Comment by sylr** [comment]
@mdvictor could you give us an update on this ?

Thank you.

**Comment by fzyzcjy** [comment]
Hi, is there any updates? Thanks!

**Comment by Marinell0** [comment]
Still a problem in Grafana v10.0.1

**Comment by mampersat** [comment]
Workaround: I added a couple unix-epoch dates to my table and it broke the gauge auto ranging for a non-date value. I modified the query to return the dates as strings and it resolved the problem.

**Comment by DrSat** [comment]
Still not fixed in v10.2

**Comment by jsaathof** [comment]
I'm running 10.0.3 and I think I managed to configure this. I have measurements from an Intel CPU where the power consumption and the thermal design are both values. I wanted to use the TDP as maximum in the graph. I have created two queries, one fo the consumption and one for the TDP.
```
SELECT mean("current_power_consumption_watts") FROM "90_days"."powerstat_package" WHERE ("host"::tag =~ /^$host$/) AND $timeFilter GROUP BY time($__interval) fill(null)
```
```
SELECT mean("thermal_design_power_watts") FROM "90_days"."powerstat_package" WHERE ("host"::tag =~ /^$host$/) AND $timeFilter GROUP BY time($__interval) fill(null)
```
By using the "config from query results" in the _Transform_ tab and using B as max will set the maximum of the graph to that value.
<img width="568" alt="Screenshot 2023-10-01 at 21 21 15" src="https://github.com/grafana/grafana/assets/864368/b986e222-de53-4f23-8b42-0414026712a5">
I have used this in a single stat. In this case the TDP is 25 W.
<img width="554" alt="Screenshot 2023-10-01 at 21 22 35" src="https://github.com/grafana/grafana/assets/864368/49395e88-282c-434e-b027-437a83664413">
Hope this helps.
